### DISCUSSION TITLE
Dummy class for Ray calls if Ray is not available?

### DISCUSSION THREAD

**Discussion by khromov** [spatie/laravel-ray#67] [Q&A]
I'm using the Ray WordPress plugin during development, but in production it's not going to be available (The plugin will be distributed through WordPress.org). Therefore I need to figure out a way to have a "dummy" `ray()` method that can be used if the real Ray framework is not available.

Here is a quick solution I came up with. It's supposed to do nothing if Ray is not available and also not throw any errors despite there being various `ray()->...` calls in the codebase. Will this fully cover Rays functionality or did I miss something?

```php
<?php
if(!function_exists('ray')) {
    class WPC_Dummy_Class {
        function __call($funName, $arguments) {
            return new WPC_Dummy_Class();
        }

        static function ray(...$args) {
            return new WPC_Dummy_Class();
        }
        
        function __get($propertyName) {
            return null;
        }

        function __set($property, $value) {
        }
    }

    function ray(...$args) {   
        return WPC_Dummy_Class::ray(...$args);
    }
}
```

**Comment by freekmurze** [comment]
For WordPress specific questions, please use the discussions tab at spatie/wordpress-ray

**Reply by khromov** [reply]
Posted here as requested:
https://github.com/spatie/wordpress-ray/discussions/9
